File: Sample Code/ObjcSampleCode/DJISdkDemo/Demo/AirLink/WiFiSSID/WiFiLinkSSIDViewController.m
//
//  WiFiLinkSSIDViewController.m
//  DJISdkDemo
//
//  Copyright © 2015 DJI. All rights reserved.
//
/**
 *  This file demonstrates how to change the SSID of DJIWiFiLink. 
 */
#import "DemoUtility.h"
#import <DJISDK/DJISDK.h>
#import "WiFiLinkSSIDViewController.h"

@interface WiFiLinkSSIDViewController ()

@end

@implementation WiFiLinkSSIDViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"WiFi SSID";
    self.rangeLabel.text = @"The input should just include alphabet, number, space, '-'\nandshould not be more than 30 characters. ";
}

-(void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    
    DJIAirLink* airLink = [DemoComponentHelper fetchAirLink];
    if (airLink && [airLink isWifiLinkSupported]) {
        [self.getValueButton setEnabled:YES];
        [self.setValueButton setEnabled:YES];
    }
    else {
        [self.getValueButton setEnabled:NO];
        [self.setValueButton setEnabled:NO];
        ShowResult(@"The product doesn't support WiFi. ");
    }
}

- (IBAction)onGetButtonClicked:(id)sender {
    __weak DJIWiFiLink* wifiLink = [DemoComponentHelper fetchAirLink].wifiLink;
    if (wifiLink) {
        WeakRef(target);
        [wifiLink getWiFiSSIDWithCompletion:^(NSString * _Nonnull ssid, NSError * _Nullable error) {
            WeakReturn(target);
            if (error) {
                ShowResult(@"ERROR: getWiFiSSID. %@", error.description);
            }
            else {
                target.getValueTextField.text = ssid;
            }
        }];
    }
}

- (IBAction)onSetButtonClicked:(id)sender {
    __weak DJIWiFiLink* wifiLink = [DemoComponentHelper fetchAirLink].wifiLink;
    if (wifiLink) {
        [wifiLink setWiFiSSID:self.setValueTextField.text withCompletion:^(NSError * _Nullable error) {
            if (error) {
                ShowResult(@"ERROR: setWiFiSSID. %@. ", error.description);
            }
            else {
                ShowResult(@"Succeed. ");
            }
        }];
    }
}


@end

File: Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Components/AirLink/WiFiSSID/WiFiLinkSSIDViewController.swift
//
//  WiFiLinkSSIDViewController.swift
//  DJISDKSwiftDemo
//
//  Copyright © 2016 DJI. All rights reserved.
//

import DJISDK
class WiFiLinkSSIDViewController: DemoGetSetViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "WiFi SSID"
        self.rangeLabel.text = "The input should just include alphabet, number, space, '-'\nandshould not be more than 30 characters. "
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        let airLink:DJIAirLink? = self.fetchAirLink()
        if airLink != nil && airLink!.isWifiLinkSupported {
            self.getValueButton.isEnabled = true
            self.setValueButton.isEnabled = true
        }
        else {
            self.getValueButton.isEnabled = false
            self.setValueButton.isEnabled = false
            self.showAlertResult("The product doesn't support WiFi. ")
        }
    }
    
    @IBAction override func onGetButtonClicked(_ sender: AnyObject) {
        
        let airLink: DJIAirLink? = self.fetchAirLink()
        if (airLink != nil) {
            let wifiLink: DJIWiFiLink? = airLink!.wifiLink
            if wifiLink != nil {
              
                wifiLink!.getWiFiSSID(completion: {[weak self](ssid: String?, error: Error?) -> Void in
                    if error != nil {
                        self?.showAlertResult("ERROR: getWiFiSSID \(error!)")
                    }
                    else {
                        self?.getValueTextField.text = ssid
                    }
                })
            }
        }
    }
    
    @IBAction override func onSetButtonClicked(_ sender: AnyObject) {
        let airLink: DJIAirLink? = self.fetchAirLink()
        if (airLink != nil) {
        let wifiLink: DJIWiFiLink? = airLink!.wifiLink
        if wifiLink != nil {
            wifiLink!.setWiFiSSID(self.setValueTextField.text!, withCompletion: {[weak self](error: Error?) -> Void in
                if error != nil {
                    self?.showAlertResult("ERROR: setWiFiSSID \(error!)")
                }
                else {
                    self?.showAlertResult("Success. ")
                }
            })
        }
        }
    }
}
